### PR TITLE
Try and better settle focus after a focused widget is removed (redux)

### PR DESCRIPTION
**Pull request by davep**
A new approach to solving #939, written to avoid the issues that https://github.com/Textualize/textual/pull/943 would have caused.

### CODE REVIEW

File: src/textual/screen.py [hunk 1]
```diff
@@ -224,19 +225,51 @@ def focus_previous(self) -> Widget | None:
...
+            self.set_focus(None)
+            return
+
+        # Now go looking for something before it, that isn't about to be
+        # removed, and which can receive focus, and go focus that.
+        chosen: DOMNode | None = None
+        for candidate in reversed(
+            focusable_widgets[widget_index + 1 :] + focusable_widgets[:widget_index]
+        ):
+            if candidate not in avoiding and candidate.can_focus:
```
Comment (willmcgugan) [inline src/textual/screen.py:267]: I think that we can be sure that `candidate.can_focus` is True, can't we?

**Review by willmcgugan — APPROVED** [review]
Yep. Looks good! Just comments.

File: src/textual/screen.py (hunk 1 above)
Comment (davep) [reply src/textual/screen.py:267]: Need a 🤦 as the real response here.

File: src/textual/app.py [hunk 2]
```diff
@@ -1419,11 +1419,16 @@ async def _on_remove(self, event: events.Remove) -> None:
         widget = event.widget
         parent = widget.parent
 
-        widget.reset_focus()
-
-        remove_widgets = widget.walk_children(
-            Widget, with_self=True, method="depth", reverse=True
+        remove_widgets = list(
```
[the remark this answers is not in the document]
Comment (davep) [reply src/textual/app.py:1422]: Aye, already noted in 11ddcdd771f54576bd14f4b4ba96ec50e2d26014 on commit.

**Pull request merged**